app/services/generate_billing.rb

```diff
@@ -0,0 +1,42 @@
+# frozen_string_literal: true
+
+class GenerateBilling
+  def initialize(month, admin)
+    @month = month
+    return unless @month
+    date = Date.today.at_beginning_of_month
```

Comment:
Global dependency (Date.today) - hidden state which is different all the time makes function not [pure](https://en.wikipedia.org/wiki/Pure_function) with consequences: hard to test and verify. 
Yes, dependency inversion/injection is also about it. Sometime it does not make sense. But usually, it's better to pass date as a parameter.
Also, what if error occurs at 31 May 11:59, and retry happen at 01 June 00:01 next day?